fastn-core/src/library2022/mod.rs

```diff
@@ -304,9 +304,42 @@ impl Library2022 {
             "user-details" => processor::user_details::process(value, kind, doc, self).await,
             "fastn-apps" => processor::apps::process(value, kind, doc, self),
             "is-reader" => processor::user_group::is_reader(value, kind, doc, self).await,
-            "sql-query" => processor::sql::process(value, kind, doc, self, "sql-query").await,
-            "sql-execute" => processor::sql::process(value, kind, doc, self, "sql-execute").await,
-            "sql-batch" => processor::sql::process(value, kind, doc, self, "sql-batch").await,
+            // send empty result when the request is for IDE previews
+            // FIXME: If the user asking for preview has write access to this site then we should not
+            // block their request.
+            "sql-query" => {
+                processor::sql::process(
+                    value,
+                    kind,
+                    doc,
+                    self,
+                    "sql-query",
+                    preview_session_id.is_none(),
+                )
+                .await
+            }
+            "sql-execute" => {
+                processor::sql::process(
+                    value,
+                    kind,
+                    doc,
+                    self,
+                    "sql-execute",
+                    preview_session_id.is_none(),
+                )
+                .await
+            }
+            "sql-batch" => {
+                processor::sql::process(
+                    value,
+                    kind,
+                    doc,
+                    self,
+                    "sql-batch",
+                    preview_session_id.is_none(),
+                )
+                .await
```

Comment:
Can't we put the complete logic here: 
```
if preview_session_id.is_none() { Default::default() } else { 
processor::sql::process(
                    value,
                    kind,
                    doc,
                    self,
                    "sql-batch",
                )
                .await
```

Reply:
Tried doing something similar in https://github.com/fastn-stack/fastn/pull/1945/commits/eb6dc4b6da57e70f3bfd48b3b099a671e106b421